executor/aggfuncs/func_rank.go

```diff
@@ -44,8 +53,13 @@ func (r *rank) ResetPartialResult(pr PartialResult) {
 
 func (r *rank) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) (memDelta int64, err error) {
 	p := (*partialResult4Rank)(pr)
-	p.rows = append(p.rows, rowsInGroup...)
-	return 0, nil
+	for _, row := range rowsInGroup {
+		p.rows = append(p.rows, row)
+		if !row.IsNull(0) {
```

Comment:
We do not need to distinguish whether it is null, they occupy the same memory space.

Reply:
I have fixed this. I use a function rowMemDeltaGens to generate memory delta, do you think it is the best way to do this?

Reply:
almost ok, `testWindowAggMemFunc` is a bit incorrect, you can refer to `testWindowFunc` implementation

Reply:
Do you mean I should use the same logic of testWindowFunc and the logic of checking memdelta of testWindowAggMemFunc which I have implemented now?

Reply:
yes, the logic is the same, we just add some memory test case.

Reply:
Maybe we don't need to repeat the logic of testWindowFunc, we just need to test the memory in testWindowAggMemFunc

Reply:
I have reformatted the code based on your advice.

Reply:
almost ok, just make a little modification